addons/udes_priorities/models/stock_move.py

```diff
@@ -39,4 +43,33 @@ def _priority_domain(self):
     def get_priorities_for_selection(self):
         Priorities = self.env["udes_priorities.priority"]
         priorities = Priorities.search(self._priority_domain())
+
+        # hard coded default value means there is always a priority to set
+        normal = self.env.ref("udes_priorities.normal")
+        priorities |= normal
+
+        for move in self.filtered(lambda m: m.priority not in priorities.mapped("reference")):
+            # Theres some race conditions around where data is available to search on this means
+            # that sometimes an invalid default can be set
+            move.priority = normal.reference
+
         return priorities.get_selection_values()
+
+    @api.constrains("priority")
+    @api.depends("priority")
+    @api.onchange("priority")
+    def _priority_cant_be_empty(self):
```

Comment:
Strictly speaking, do we need all of constrains, depends and onchange here? Would we get what we need with just constrains?

Reply:
constrains and onchange are needed, I think depends is the one we can remove... but not 100%

Reply:
removed :)